### PR TITLE
Format Preserving Encryption

### PR DESCRIPTION
**Pull request by jarrettranderson**
Add a privacy provider that applies a format-preserving encryption (FPE) algorithm on target data values.

### CODE REVIEW

**Review by dumashwang — APPROVED** [review]
Looks good to me.

File: de-identification-app/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ###############################################################################
-# (C) Copyright IBM Corp. 2016,2020
+# (C) Copyright IBM Corp. 2016,2021
 #
 # SPDX-License-Identifier: Apache-2.0
 ###############################################################################
  
-
 logging.level.root=INFO
```
Comment (DenisRicard) [inline de-identification-app/src/main/resources/application.properties:7]: Should INFO be the default ? You'd think that this is going to log PHI at one place or another

Comment (jarrettranderson) [reply de-identification-app/src/main/resources/application.properties:7]: INFO is what we've been testing primarily.  It seems reasonable.  The problems we've seen have been at DEBUG level except for the new third-party code.   We can change it to WARN if you are more comfortable with that?

**Review by DenisRicard — APPROVED** [review]
Very good. Hard to find flaws even trying to be picky :-)

File: de-identification-app/src/main/resources/application.properties (hunk 1 above)
Comment (DenisRicard) [reply de-identification-app/src/main/resources/application.properties:7]: DEBUG would be really bad as default, yes. If you look at the logs at INFO after a few unit tests ran for example and it seems reasonable, we can simply document that for production we recommend Warn or Error. If we don't trust INFO anymore as default then we could go to WARN in that file

Comment (jarrettranderson) [reply de-identification-app/src/main/resources/application.properties:7]: For now, I would say we don't have a reason not to trust INFO.

**Pull request merged**